tests/h/schemas/base_test.py

```diff
@@ -87,6 +116,90 @@ def test_it_returns_all_errors_in_message(self):
         assert message.startswith("'foo' is a required property, 'bar' is a required property")
 
 
+class TestRemoveUnknownProperties(object):
+
+    @pytest.mark.parametrize('data,expected', [
+        (
+            # data
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string"
+                }
+             }},
+            # expected
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string"
+                }
+             }},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "no_exist": "string",
+             "baz": {
+                "foo": "string",
+                "no_exist": "string",
+                "bar": {
+                    "foo": "string",
+                    "no_exist": "string",
+                }
+             }},
+            # expected
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string",
+                }
+             }},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "no_exist": {
+                "bar": "string",
+             }},
+            # expected
+            {"foo": "string"},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "bar": {
+                "no_exist": "string",
+             }},
+            # expected
+            {"foo": "string"},
+
+        ),
+        (
+            # data
+            MultiDict([("foo", "string"),
+                       ("foo", "str"),
+                       ("baz", {"foo": "string"}),
+                       ("no_exist", {"bar": "string"}),
+                       ("no_exist", "string")]),
+            # expected
+            MultiDict([("foo", "string"),
+                       ("foo", "str"),
+                       ("baz", {"foo": "string"})]),
+
+    )])
+    def test_it_removes_unknown_properities_from_data(self, data, expected):
```

Comment:
Typo "properities".

Reply:
There are certain words that I kept swizzling letters on when I was writing this-I was having the worst time with "unknown" I kept leaving out "n"s so I'd type "uknown" or "unkown". Typing is hard. lol